Compilation Error on .NET Framework

### Describe the bug

When using .NET Framework 4.8 project build fails with the following errors:
<details>
<summary>Errors Log</summary>
...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullAttribute.g.cs(12,27): error CS0101: The namespace 'System.Diagnostics.CodeAnalysis' already contains a definition for 'NotNullAttribute'


...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.g.cs(11,27): error CS0101: The namespace 'System.Diagnostics.CodeAnalysis' already contains a definition for 'NotNullIfNotNullAttribute'


...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.g.cs(10,6): error CS0579: Duplicate 'AttributeUsage' attribute


...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullAttribute.g.cs(11,6): error CS0579: Duplicate 'AttributeUsage' attribute


...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.g.cs(17,16): error CS0111: Type 'NotNullIfNotNullAttribute' already defines a member called 'NotNullIfNotNullAttribute' with the same parameter types
</details>

Everything is fine when using .NET 6, I guess the code generator is not compatible with .NET Framework when I have `<Nullable>enable</Nullable>` and `<LangVersion>9</LangVersion>` but not sure, needs to be tested.

### Regression

_No response_

### Steps to reproduce

```text
1. Create a .NET Framework 4.8 project with `<Nullable>enable</Nullable>` and `<LangVersion>9</LangVersion>`.
2. Add CommunityToolkit.Mvvm 8.0.0 NuGet package
3. Try to build it.
```


### Expected behavior

The app should get built without any problem.

### Screenshots

_No response_

### IDE and version

VS 2022

### IDE version

17.3.1

### Nuget packages

- [ ] CommunityToolkit.Common
- [ ] CommunityToolkit.Diagnostics
- [ ] CommunityToolkit.HighPerformance
- [X] CommunityToolkit.Mvvm (aka MVVM Toolkit)

### Nuget package version(s)

8.0.0

### Additional context

_No response_

### Help us help you

No, just wanted to report this

Hmm, I think I hit this same thing with 8.1-preview1, the error messages are basically the same:

```
2>I:\code\...\...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullAttribute.g.cs(12,27,12,43): error CS0101: The namespace 'System.Diagnostics.CodeAnalysis' already contains a definition for 'NotNullAttribute'
2>I:\code\...\...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.g.cs(11,27,11,52): error CS0101: The namespace 'System.Diagnostics.CodeAnalysis' already contains a definition for 'NotNullIfNotNullAttribute'
2>I:\code\...\...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullAttribute.g.cs(11,6,11,20): error CS0579: Duplicate 'AttributeUsage' attribute
2>I:\code\...\...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.g.cs(10,6,10,20): error CS0579: Duplicate 'AttributeUsage' attribute
2>I:\code\...\...\CommunityToolkit.Mvvm.SourceGenerators\CommunityToolkit.Mvvm.SourceGenerators.NullabilityAttributesGenerator\System.Diagnostics.CodeAnalysis.NotNullIfNotNullAttribute.g.cs(17,16,17,41): error CS0111: Type 'NotNullIfNotNullAttribute' already defines a member called 'NotNullIfNotNullAttribute' with the same parameter types
```

But I have a .NET Standard 2.0 class library with language set to `latest`.  That project seems to build. But it's the UWP app it's being referenced in that throws these errors...

--- a/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
+++ b/src/CommunityToolkit.Mvvm/CommunityToolkit.Mvvm.targets
@@ -41,6 +41,36 @@
     <Warning Condition ="'$(MVVMToolkitCurrentCompilerVersionIsNotNewEnough)' == 'true'" Text="The MVVM Toolkit source generators have been disabled on the current configuration, as they need Roslyn 4.x in order to work. The MVVM Toolkit will work just fine, but features relying on the source generators will not be available."/>
   </Target>
 
+  <!--
+    Manually remove duplicate analyzers if Roslyn component versioning is not supported (ie. if a legacy .csproj project is used).
+    This target is only run if Roslyn 4.0 or greater is present, as otherwise all analyzers would have already been removed anyway.
+  -->
+  <Target Name="MVVMToolkitRemoveDuplicateAnalyzersWhenRoslynComponentVersioningIsNotSupported"
+          Condition="'$(MVVMToolkitCurrentCompilerVersionIsNotNewEnough)' != 'true' AND '$(SupportsRoslynComponentVersioning)' != 'true'"
+          AfterTargets="ResolvePackageDependenciesForBuild;ResolveNuGetPackageAssets"
+          DependsOnTargets="MVVMToolkitRemoveAnalyzersForRoslyn3">
+
+      <!--
+        This switch manually implements Roslyn component versioning. That is, it checks the current version of Roslyn and
+        removes and removes all analyzers except the highest version that is supported. The fallback is just Roslyn 4.0.
+      -->
+    <PropertyGroup>
+      <MVVMToolkitSelectedRoslynAnalyzerDirectoryName Condition="$([MSBuild]::VersionGreaterThanOrEquals($(MVVMToolkitCurrentCompilerVersion), 4.3))">roslyn4.3</MVVMToolkitSelectedRoslynAnalyzerDirectoryName>
+      <MVVMToolkitSelectedRoslynAnalyzerDirectoryName Condition="'$(MVVMToolkitSelectedRoslynAnalyzerDirectoryName)' == ''">roslyn4.0</MVVMToolkitSelectedRoslynAnalyzerDirectoryName>
+    </PropertyGroup>
+    <ItemGroup>
+      
+      <!--
+        This condition is a bit convoluted, but it's essentially just selecting all analyzers from the NuGet package that don't have the target Roslyn directory name in their full path.
+        For instance, if Roslyn 4.3 is the highest supported version, the target directory name will be "roslyn 4.3", and this condition will filter out all analyzers with a path such
+        as: "C:\...\.nuget\...\CommunityToolkit.Mvvm\analyzers\roslyn4.0\cs\CommunityToolkit.Mvvm". The [System.String]::Concat trick is used to achieve two things: we can't directly
+        invoke a property function (ie. Contains in this case) on a metadata item, so we need an intermediate string to invoke it on. We could also use [System.String]::new, but using
+        Concat is more efficient as it'll just skip the allocation entirely if one of the two inputs is an empty string, which is the case here.
+      -->
+      <Analyzer Remove="@(MVVMToolkitAnalyzer)" Condition="!$([System.String]::Concat('', '%(MVVMToolkitAnalyzer.FullPath)').Contains('$(MVVMToolkitSelectedRoslynAnalyzerDirectoryName)'))"/>
+    </ItemGroup>
+  </Target>
+
   <!-- Remove the analyzer if Roslyn is missing -->
   <Target Name="MVVMToolkitRemoveAnalyzersForRosynNotFound"
           Condition="'$(CSharpCoreTargetsPath)' == ''"

